### ISSUE TITLE
bug: Passing component state using React Router's `location.state` triggers `no-event-handler` rule

### ISSUE THREAD

**Issue by vitalyster** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#44]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.5.5

### ESLint version

9.34.0

### Describe the bug

I'm using `useEffect` to conditionally fetch initial component state from network when it is not present in [`location.state`](https://api.reactrouter.com/v7/interfaces/react_router.Location.html#state-1)
There is no any "event" to handle, so it is a false positive.

### Steps To Reproduce

Something like this:
```
function MyRoute = () => {
  const { state } = useLocation()
  const [data, setData] = useState(state.data)
  useEffect(() => {
    if (!data) {
       setData(loadDataFromAPI())
    }
  }, [data])
  return <p>${data}</p>
}
```

When I navigate programmatically to MyRoute I pass required data from previous location, but there is no previous location when I opened link to MyRoute directly, so I'm unable to lift the state up

### Expected Behavior

It should not trigger `no-event-handler` route
